ably/realtime_channel.go

```diff
@@ -585,25 +585,49 @@ func (em ChannelEventEmitter) OffAll() {
 }
 
 // Publish publishes a single message to the channel with the given event name and message payload.
-// A callback may optionally be passed in to this call to be notified of success or failure of the operation.
-// When publish is called with this client library, it won't attempt to implicitly attach to the channel,
-// so long as transient publishing is available in the library. Otherwise, the client will implicitly attach (RTL6i).
 //
-// This implicitly attaches the channel if it's not already attached. If the
-// context is canceled before the attach operation finishes, the call
-// returns with an error, but the operation carries on in the background and
-// the channel may eventually be attached and the message published anyway.
+// This will block until either the publish is acknowledged or fails to deliver.
+//
+// If the context is cancelled before the attach operation finishes, the call
+// returns an error but the publish will carry on in the background and may
+// eventually be published anyway.
 func (c *RealtimeChannel) Publish(ctx context.Context, name string, data interface{}) error {
 	return c.PublishMultiple(ctx, []*Message{{Name: name, Data: data}})
 }
 
+// PublishAsync is the same as Publish except instead of blocking it calls onAck
+// with nil if the publish was successful or the appropriate error.
+//
+// Note onAck must not block as it will block the event loop.
+func (c *RealtimeChannel) PublishAsync(name string, data interface{}, onAck func(err error)) error {
+	return c.PublishMultipleAsync([]*Message{{Name: name, Data: data}}, onAck)
+}
+
 // PublishMultiple publishes all given messages on the channel at once.
 //
-// This implicitly attaches the channel if it's not already attached. If the
-// context is canceled before the attach operation finishes, the call
-// returns with an error, but the operation carries on in the background and
-// the channel may eventually be attached and the message published anyway (RTL6i).
+// If the context is cancelled before the attach operation finishes, the call
+// returns an error but the publish will carry on in the background and may
+// eventually be published anyway.
 func (c *RealtimeChannel) PublishMultiple(ctx context.Context, messages []*Message) error {
+	listen := make(chan error, 1)
+	onAck := func(err error) {
+		listen <- err
+	}
+	if err := c.PublishMultipleAsync(messages, onAck); err != nil {
+		return err
+	}
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-listen:
+		return err
+	}
+}
+
+// PublishMultipleAsync is the same as PublishMultiple except it calls onAck instead of blocking
```

Comment:
`PublishMultipleAsync` is the same as `PublishAsync`, only different is, multiple messages can be published.

Reply:
I think we're bikeshedding a bit here slightly rewording comments?